sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.controller.submitApplication.e2e-spec.ts

```diff
@@ -0,0 +1,364 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import { TestingModule } from "@nestjs/testing";
+import * as request from "supertest";
+import { DataSource } from "typeorm";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  FakeStudentUsersTypes,
+  getStudentToken,
+  mockUserLoginInfo,
+} from "../../../testHelpers";
+import {
+  createE2EDataSources,
+  E2EDataSources,
+  saveFakeStudent,
+  saveFakeApplication,
+  ensureProgramYearExists,
+  createFakeApplication,
+  getProviderInstanceForModule,
+  saveFakeSFASIndividual,
+  createFakeEducationProgramOffering,
+  createFakeUser,
+} from "@sims/test-utils";
+import {
+  Application,
+  ApplicationStatus,
+  EducationProgramOffering,
+  OfferingIntensity,
+  ProgramYear,
+  Student,
+} from "@sims/sims-db";
+import { addDays, getISODateOnlyString } from "@sims/utilities";
+import { SaveApplicationAPIInDTO } from "../models/application.dto";
+import { SystemUsersService } from "@sims/services";
+import { FormNames, FormService } from "../../../services";
+import { AppStudentsModule } from "../../../app.students.module";
+import { createFakeSFASPartTimeApplication } from "@sims/test-utils/factories/sfas-part-time-application";
+
+describe("ApplicationStudentsController(e2e)-submitApplication", () => {
+  let app: INestApplication;
+  let appModule: TestingModule;
+  let appDataSource: DataSource;
+  let sharedStudent: Student;
+  let db: E2EDataSources;
+  let programYear: ProgramYear;
+  let systemUsersService: SystemUsersService;
+  let formService: FormService;
+  const PROGRAM_YEAR_PREFIX = 1990;
+
+  beforeAll(async () => {
+    const { nestApplication, module, dataSource } =
+      await createTestingAppModule();
+    app = nestApplication;
+    appModule = module;
+    appDataSource = dataSource;
+    db = createE2EDataSources(dataSource);
+    sharedStudent = await saveFakeStudent(db.dataSource);
+    systemUsersService = nestApplication.get(SystemUsersService);
+    // Program Year for the following tests.
+    programYear = await ensureProgramYearExists(db, PROGRAM_YEAR_PREFIX);
```

Comment:
Is there a need for these tests to run in a particular program year, if not i would recommend to use the default program year in factory.